### PR TITLE
Documentation for ingester instance limits in runtime configuration

### PR DESCRIPTION
**Pull request by 56quarters**
Signed-off-by: Nick Pillitteri <nick.pillitteri@grafana.com>

**Checklist**
- [ ] Tests updated
- [X] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/configuration/arguments.md [hunk 1]
```diff
@@ -404,7 +404,7 @@ It also talks to a KVStore and has it's own copies of the same flags used by the
 
 Cortex has a concept of "runtime config" file, which is simply a file that is reloaded while Cortex is running. It is used by some Cortex components to allow operator to change some aspects of Cortex configuration without restarting it. File is specified by using `-runtime-config.file=<filename>` flag and reload period (which defaults to 10 seconds) can be changed by `-runtime-config.reload-period=<duration>` flag. Previously this mechanism was only used by limits overrides, and flags were called `-limits.per-user-override-config=<filename>` and `-limits.per-user-override-period=10s` respectively. These are still used, if `-runtime-config.file=<filename>` is not specified.
 
-At the moment, two components use runtime configuration: limits and multi KV store.
+At the moment, three components use runtime configuration: per-user limits, multi KV store, and ingester instance limits.
```
Comment (pstibrany) [inline docs/configuration/arguments.md:407]: Shall we reword this to avoid "components"? For me, "component" means ingester or compactor.

What about: "At the moment runtime configuration may contain per-user limits, configuration for multi KV store and ingester instance limits."

Comment (56quarters) [reply docs/configuration/arguments.md:407]: Good idea, will do!

File: docs/configuration/arguments.md [hunk 2]
```diff
@@ -506,6 +510,38 @@ Valid per-tenant limits are (with their corresponding flags for default values):
 
    Requires `-distributor.replication-factor`, `-distributor.shard-by-all-labels`, `-distributor.sharding-strategy` and `-distributor.zone-awareness-enabled` set for the ingesters too.
 
+## Ingester Instance Limits
+
+Cortex ingesters support limits that are applied per-instance, meaning they apply to each ingester process. These can be used to ensure individual ingesters are not overwhelmed regardless of any per-user limits. These limits can be set under the `ingester.instance_limits` block in the global configuration file, with command line flags, or under the `ingester_limits` field in the runtime configuration file.
```
[the remark this answers is not in the document]
Comment (56quarters) [reply docs/configuration/arguments.md:515]: Sure, will do.

**Review by pstibrany — APPROVED** [review]
Thank you! This is very useful.

**Review by pracucci — APPROVED** [review]
Amazing, thanks!
